source/magpie/dataframe.d

```diff
@@ -767,6 +767,90 @@ public:
         indx.optimize();
     }
 
+    /++
+    from_csv rebuild for faster read
+    +/
+    void fastCSV(string path, size_t indexDepth, size_t columnDepth, char sep = ',')
+    {
+        import std.array: appender, split;
+        import std.stdio: File;
+        import std.string: chomp;
+
+        File csvfile = File(path, "r");
+        string[][] lines;
+        int totalLines = 0;
+
+        while(!csvfile.eof())
+        {
+            ++lines.length;
+            lines[totalLines++] = chomp(csvfile.readln()).split(sep);
+        }
+
+        indx.row.titles.length = indexDepth;
+        indx.row.index.length = indexDepth;
+        indx.column.index.length = columnDepth;
+        indx.row.codes.length = indexDepth;
+        indx.column.codes.length = columnDepth;
+        indx.row.titles = lines[columnDepth - 1][0 .. indexDepth];
+
+        foreach(i, ele; lines[0 .. columnDepth])
+            indx.column.index[i] = ele[indexDepth .. $];
+
+        foreach(i, ele; lines[columnDepth .. $])
+        {
+            if(ele.length > 0)
```

Comment:
for conditionals like this you can reduce the indenting by writing
```
if(ele.length == 0)
    continue;
```

Reply:
Done in dec86d1
I'll eventually scrub the entire code base looking for cases like this :+1: